### DISCUSSION TITLE
Repo defaults

### DISCUSSION THREAD

**Discussion by barisunsalhn** [gogs/gogs#7037] [Proposal]
I'm considering to take [this](https://github.com/gogs/gogs/issues/6881) issue. Is it appropriate? If yes, what should I know before starting? Any suggestions? Should I create repo_defaults table ( columns will be CreateRepoOptions's struct's elements)inside database?

**Comment by unknwon** [comment]
Hey, thanks for asking!

Here are the steps I can think of top off my head:

1. Create the `repo_default` table (so far our convention is to use singular table name), columns will generally be CreateRepoOptions's field as you suggested, and at most one row should exist.
2. Load the defaults to the page for creating and migrating repositories.
